### ISSUE TITLE
[Feature Request] Replace some character

### ISSUE THREAD

**Issue by toshiya14** [Soreepeong/FFXIV-FontChanger#5]
Some symbols or punctuations are not displays very well and in this case, I would like to change the symbol to another.
For example:

![图片](https://user-images.githubusercontent.com/13333533/171086884-9ffa034d-093a-4cf7-b832-0e656724b18b.png)

Replace the following characters while genereting.

"《" -> "「"
"》" -> "」"

**Comment by Soreepeong** [comment]
That shouldn't be too difficult to fix.

Regardless, to fix that kind of problem, you will have to adjust horizontal offset. Note the `(람)` here.

Without it:
![image](https://user-images.githubusercontent.com/3614868/171086979-a4478abc-2ce9-4e82-a461-0064f791df1c.png)

With it:
![image](https://user-images.githubusercontent.com/3614868/171086999-d4f2730b-4e0d-49fb-931a-6939aa90a0eb.png)

**Comment by toshiya14** [comment]
I changed the horizontal offset to `2px`, but nothing changes.

**Comment by toshiya14** [comment]
![图片](https://user-images.githubusercontent.com/13333533/171089160-0488491c-d857-48d1-ac0d-341c6e4d2d5c.png)

If I set to `8px`, it changes a little.
But I think the right side would be overflow, if I continue to increase the offset. XD

**Comment by Soreepeong** [comment]
Try this one, and let me know if it works as you would think.

[XivRes.FontGenerator64 (2).zip](https://github.com/Soreepeong/FFXIV-FontChanger/files/8801915/XivRes.FontGenerator64.2.zip)

For now, though, you'll have to edit the saved json files manually; save it, open the file with a text editor, and edit `codepointReplacements`.
```json
{
    "codepointReplacements": {"《":"「", "》":"」"},`
}
```

**Comment by toshiya14** [comment]
Could you give me a new release? I did not have the environment for cpp ready.
So I think I could hardly compile it by myself.

**Comment by toshiya14** [comment]
> Try this one, and let me know if it works as you would think.
> 
> [XivRes.FontGenerator64 (2).zip](https://github.com/Soreepeong/FFXIV-FontChanger/files/8801915/XivRes.FontGenerator64.2.zip)
> 
> For now, though, you'll have to edit the saved json files manually; save it, open the file with a text editor, and edit `codepointReplacements`.
> 
> ```json
> {
>     "codepointReplacements": {
>         "《": "「",
>         "》": "」"
>     }
> }
> ```

I am sorry, I did not realize you posted the release.

**Comment by toshiya14** [comment]
The version you have just provided loaded the config as `ascent` always `11`.
![图片](https://user-images.githubusercontent.com/13333533/171092648-bbdadb9f-b95f-42a9-9b9e-c29bc5743b87.png)

**Comment by toshiya14** [comment]
I found that when I load the json, the chnasix becomes empty.

**Comment by Soreepeong** [comment]
That is correct. That font features 14px line height and 11px ascent when font size is set to 12px. 

For everything being displayed in the same line, each font face will be configured from the first entry in the list. In this case, it'll take line height and ascent as "0px" from the first entry.

**Comment by toshiya14** [comment]
![图片](https://user-images.githubusercontent.com/13333533/171096593-2d55caaf-1e9f-4933-84d9-1ae714bdff62.png)

**Comment by Soreepeong** [comment]
Can you share the latest file you're working with again?

**Comment by toshiya14** [comment]
The gist has been updated.
https://gist.github.com/toshiya14/f4d255152b398194bbd6c006e0d25530

This version with release 0.0.0.1 is fine.
But just a little problem.
![图片](https://user-images.githubusercontent.com/13333533/171096955-6ea5bc39-f626-4367-8518-108e25925eb0.png)

**Comment by toshiya14** [comment]
> Try this one, and let me know if it works as you would think.
> 
> [XivRes.FontGenerator64 (2).zip](https://github.com/Soreepeong/FFXIV-FontChanger/files/8801915/XivRes.FontGenerator64.2.zip)
> 
> For now, though, you'll have to edit the saved json files manually; save it, open the file with a text editor, and edit `codepointReplacements`.
> 
> ```json
> {
>     "codepointReplacements": {
>         "《": "「",
>         "》": "」"
>     }
> }
> ```

This version seems not load my preset correctly.
Then generated mod is works bad.

**Comment by Soreepeong** [comment]
Ah, it will do that if game path has not been configured correctly, and it seems that you've downloaded the updated version but didn't change config.json. Try copying that file over to the new exe file directory.

**Comment by toshiya14** [comment]
Just an advice. You could add the version number to the windows title.
If I post screenshot, then you could identify them. XD

**Comment by toshiya14** [comment]
This settings seems not work, I am using "UTF-8" to saving my preset.
Should I use a native language encoding such as "ANSI/GBK"?
![图片](https://user-images.githubusercontent.com/13333533/171100064-ac095e9f-b88d-49a5-8a29-c35613800f40.png)
![图片](https://user-images.githubusercontent.com/13333533/171100117-d4d2246c-fa5c-4d78-955b-721644973f49.png)

**Comment by Soreepeong** [comment]
Ah, I should have said better. It goes inside face element, like this:

![image](https://user-images.githubusercontent.com/3614868/171100444-65acd217-8b1b-45d5-a82c-2775f2ec41c7.png)

Also yeah, only UTF-8 is the valid JSON charset!

**Comment by toshiya14** [comment]
![图片](https://user-images.githubusercontent.com/13333533/171101207-aa00b11e-b051-436e-b18a-46e390faf537.png)

I tried to add the replacement. It becomes more strange than before. XD

I have tried to input 《》 in chat, and others could see the them, but I could not. 
![图片](https://user-images.githubusercontent.com/13333533/171101491-98cf3fdd-2a39-4f0e-b2d6-0edd44493c33.png)
It should be `《》←左边有东西吗？`

**Comment by Soreepeong** [comment]
[XivRes.FontGenerator64 (2).zip](https://github.com/Soreepeong/FFXIV-FontChanger/files/8802528/XivRes.FontGenerator64.2.zip)

I've fixed that problem. Try it again! I can now confirm that at least in preview it looks good.
![image](https://user-images.githubusercontent.com/3614868/171106095-f4cc30da-7a76-426d-82a5-1559197acb36.png)

**Comment by toshiya14** [comment]
![图片](https://user-images.githubusercontent.com/13333533/171114655-27b02bf9-5c0a-43af-a15f-baa82cfbc02d.png)
It works!

But the quotes still strange.
I have tested the preset you have posted in another issue.
That works fine.
https://github.com/Soreepeong/FFXIV-FontChanger/issues/2#issuecomment-1141696064
